File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueTypeChangedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.Issue;

import java.time.Instant;

public record IssueTypeChangedEvent(
        long projectId,
        long issueId,
        long updaterId,
        String oldType,
        String newType,
        Instant changedAtUtc
) {
    public static IssueTypeChangedEvent from(Issue issue, long updaterId, String oldType) {
        return new IssueTypeChangedEvent(
                issue.getProjectId(),
                issue.getId(),
                updaterId,
                oldType,
                issue.getType().name(),
                Instant.now()
        );
    }
}

File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueTypeChangedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueTypeChangedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("updaterId")]
    public long UpdaterId { get; set; }

    [JsonPropertyName("oldType")]
    public string OldType { get; set; } = string.Empty;

    [JsonPropertyName("newType")]
    public string NewType { get; set; } = string.Empty;

    [JsonPropertyName("changedAtUtc")]
    public DateTime ChangedAtUtc { get; set; } = DateTime.UtcNow;
}